src/MainWindow.vala

```diff
@@ -7,6 +7,25 @@ public class Music.MainWindow : Hdy.ApplicationWindow {
     construct {
         Hdy.init ();
 
+        var queue_header = new Hdy.HeaderBar () {
+            hexpand = true,
+            show_close_button = true
+        };
+
+        unowned var queue_header_context = queue_header.get_style_context ();
+        queue_header_context.add_class (Granite.STYLE_CLASS_DEFAULT_DECORATION);
+        queue_header_context.add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var queue_listbox = new Gtk.ListBox () {
+            expand = true,
+            sensitive = false
+        };
+        queue_listbox.bind_model (PlaybackManager.get_default ().queue_liststore, create_queue_row);
+
+        var queue = new Gtk.Grid ();
```

Comment:
This should probably have view class. Looks like headerbar inside view or even headerbar.view doesn't actually work